### ISSUE TITLE
Memory leak while using Camera 

### ISSUE THREAD

**Issue by ozc** [kivy/kivy#3872]
While using Camera and setting play to true, memory usage keeps increasing. When memory usage hits above 90% for the python process, and that's 100% for the system, camera shows 1 frame per 10-20 seconds.

I'm using Kivy 1.9.1-dev and Python v2.7.9 on Raspberry Pi 2, Raspian jessie (Release 8.0). Camera is a Logitech C920

[    7.068333] uvcvideo: Found UVC 1.00 device HD Pro Webcam C920 (046d:082d)
[    7.068341] usbcore: registered new interface driver snd-usb-audio
[    7.070633] input: HD Pro Webcam C920 as /devices/platform/soc/3f980000.usb/usb1/1-1/1-1.4/1-1.4:1.0/input/input2
[    7.071224] usbcore: registered new interface driver uvcvideo
[    7.071242] USB Video Class driver (1.1.1)

I have checked the camera example (http://kivy.org/docs/examples/gen__camera__main__py.html) in case there's a problem with my code, and same issue happens there too.

**Comment by thecodingwizard** [comment]
Similar thing is happening to me: I'm loading a live-updating image (taken from raspberry pi cam) 30 times per second and memory steadily increases. PS: This is a Ubuntu VM getting an image from a raspberry pi.

Here's the code:

intercom.py

    import time

    from kivy.app import App
    from kivy.clock import Clock
    from kivy.core.image import Image
    from kivy.loader import Loader
    from kivy.properties import ObjectProperty
    from kivy.uix.widget import Widget


    class IntercomImage(Widget):
        image = ObjectProperty(Image)
        fps = 30.0

       def update(self, dt):
            proxyImg = Loader.image("http://192.168.15.113/cam/cam_pic.php?time=" + str(time.time()))
            proxyImg.bind(on_load=self.image_loaded)

        def image_loaded(self, proxyImage):
            if proxyImage.image.texture:
                #self.image.texture = proxyImage.image.texture
                pass



    class IntercomApp(App):
        def build(self):
            image = IntercomImage()
            #image.update(0)
            Clock.schedule_interval(image.update, 1.0/30.0)
            return image


    if __name__ == "__main__":
        IntercomApp().run()

`intercom.kv`

    #kivy 1.9.1
    
    <IntercomImage>:
        image: cam_img
        Image:
            id: cam_img
            width: 1024
            height: 576

perhaps `proxyImage` from `image_loaded` is not getting garbage collected?

**Comment by riataman** [comment]
Hey, I ran into this. I'm not sure why this was closed as the other people commenting seem to be talking about a completely different case than the original reporter.

The issue with the memory leak in the raspberry pi is easily demonstrated by running  /examples/camera/main.py in the raspberry pi running the latest raspbian jessie.

The memory leaks very fast and in my system with a Logitech C920 the memory would be completely gonne in less than a minute.

I found that if I change here: https://github.com/kivy/kivy/blob/master/kivy/core/camera/camera_opencv.py#L137 The format from bgr to rgb then the memory leak is fixed. But the colors are all wrong.
